tools/topology/topology1/sof-tgl-max98357a-rt5682.m4

```diff
@@ -426,24 +448,31 @@ ifelse(
                 SSP_CLOCK(fsync, 48000, codec_slave),
                 SSP_TDM(2, 16, 3, 3),
                 SSP_CONFIG_DATA(SSP, SPK_SSP_INDEX, 16)))',
+	CODEC, `CS35L41', `
+	SSP_CONFIG(DSP_A, SSP_CLOCK(mclk, 19200000, codec_mclk_in),
+		SSP_CLOCK(bclk, 6144000, codec_slave),
+		SSP_CLOCK(fsync, 48000, codec_slave),
+		SSP_TDM(4, 32, 3, 15),
+		SSP_CONFIG_DATA(SSP, SPK_SSP_INDEX, 24)))',
 	)')
 
+ifdef(`NO_HEADPHONE',`',`
 # SSP 0 (ID: 0)
 DAI_CONFIG(SSP, 0, 0, SSP0-Codec,
 	SSP_CONFIG(I2S, SSP_CLOCK(mclk, 19200000, codec_mclk_in),
 		SSP_CLOCK(bclk, 2400000, codec_slave),
 		SSP_CLOCK(fsync, 48000, codec_slave),
 		SSP_TDM(2, 25, 3, 3),
-		SSP_CONFIG_DATA(SSP, 0, 24, 0, 0, 0, SSP_CC_BCLK_ES)))
+		SSP_CONFIG_DATA(SSP, 0, 24, 0, 0, 0, SSP_CC_BCLK_ES)))')
 
 # 4 HDMI/DP outputs (ID: 3,4,5,6)
-DAI_CONFIG(HDA, 0, 3, iDisp1,
+DAI_CONFIG(HDA, 0, HDMI_BE_ID_BASE, iDisp1,
 	HDA_CONFIG(HDA_CONFIG_DATA(HDA, 0, 48000, 2)))
-DAI_CONFIG(HDA, 1, 4, iDisp2,
+DAI_CONFIG(HDA, 1, eval(HDMI_BE_ID_BASE + 1), iDisp2,
```

Comment:
i wonder if we should just make this its own M4 we can include with a parameter to set the base, would save some code space for all boards.

Reply:
yeah, having a separate M4 file that could be included here with the ID definitions for each board would make the maintenance easier.